### PR TITLE
Fix reset TensorRunningAccum

### LINKED ISSUE #5073: TensorRunningAccum does not reset as it assumed to be
## 🐛 Bug
I looked into the source code in [this file](https://github.com/PyTorchLightning/pytorch-lightning/blob/7e8673debd2f8bc8ae639063c4aee71f774fc304/pytorch_lightning/trainer/supporters.py#L51) and found this snippet that does not make sense to me.

```python
    def reset(self) -> None:
        """Empty the accumulator."""
        self = TensorRunningAccum(self.window_length)
```

### To Reproduce
I made a short code snippet to verify this behavior as follows.

```python
from pytorch_lightning.trainer.supporters import TensorRunningAccum

running_accum = TensorRunningAccum(window_length=10)
running_accum.current_idx = 100     # just want to change the value
print(id(running_accum))
print(running_accum.current_idx)

running_accum.reset()
print(running_accum.current_idx)    # current_idx is expected to be 0
print(id(running_accum))
```
And the result is
```
139686191621984
100
100
139686191621984
```

### PR DESCRIPTION
**Pull request by VinhLoiIT**
## What does this PR do?
Fixes #5073

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

File: pytorch_lightning/trainer/supporters.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def __init__(self, window_length: int):
 
     def reset(self) -> None:
         """Empty the accumulator."""
-        self = TensorRunningAccum(self.window_length)
+        self.__init__(self.window_length)
```
Comment (tchaton) [inline pytorch_lightning/trainer/supporters.py:53]: Could you add a test to make sure `TensorRunningAccum ` is having the right behaviour ?

Comment (VinhLoiIT) [reply pytorch_lightning/trainer/supporters.py:53]: Sure. I've tested it manually, but I would add the test in the next commit as you suggested.

Comment (carmocca) [reply pytorch_lightning/trainer/supporters.py:53]: I don't think we have tests for `TensorRunningAccum` at all, so you'll have to create them

**Review by awaelchli — APPROVED** [review]
verified fails on master

**Pull request merged**